server/api/migrations/versions/650f0ce2da6f_add_alert_history.py

```diff
@@ -0,0 +1,130 @@
+        sa.Column(
+            "activation_time", mysql.DATETIME(timezone=True, fsp=3), nullable=False
+        ),
+        sa.Column(
+            "name", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "project", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("data", sa.JSON(), nullable=True),
+        sa.Column(
+            "entity_id", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "entity_kind",
+            sa.String(length=255, collation="utf8mb3_bin"),
+            nullable=False,
+        ),
+        sa.Column(
+            "event_kind", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "severity", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("number_of_events", sa.Integer(), nullable=False),
+        sa.PrimaryKeyConstraint("activation_time", "id", name="_alert_history_uc"),
+    )
+    op.create_index(
+        "ix_alert_history_activation_time",
+        "alert_history",
+        ["activation_time"],
+        unique=False,
+    )
+    op.create_index(
+        "ix_alert_history_project_name",
+        "alert_history",
+        ["project", "name"],
+        unique=False,
+    )
+    # ### end Alembic commands ###
+
+    partition_interval = os.getenv("PARTITION_INTERVAL", "YEARWEEK").upper()
+    valid_partition_intervals = ["DAY", "YEARWEEK", "MONTH"]
+
+    # Validate the partition interval
+    if partition_interval not in valid_partition_intervals:
+        raise ValueError(
+            f"Partition interval can only be one of the following: {valid_partition_intervals}"
+        )
+
+    # Current UTC date
+    now_utc = datetime.utcnow()
+
+    # Define partition name format based on interval
+    partition_name_format_mapping = {
+        "YEARWEEK": "%Y%V",  # ISO week number (for weeks starting Monday)
+        "MONTH": "%Y%m",  # Format as 'YYYYMM' (year and month)
+        "DAY": "%Y%m%d",  # Format as 'YYYYMMDD' (year, month, day)
+    }
+
+    # Generate the partition name
+    partition_name = now_utc.strftime(partition_name_format_mapping[partition_interval])
+
+    # Calculate partition value depending on interval
+    if partition_interval == "YEARWEEK":
+        partition_value = int(
+            now_utc.strftime("%Y%V")
+        )  # Example: 202444 for 44th week of 2024
+        partition_expression = "YEARWEEK(activation_time, 1)"
+    else:
+        partition_value = now_utc.strftime("%Y-%m-%d")
+        partition_expression = f"{partition_interval}(activation_time)"
+
+    # Construct SQL for partitioning
+    partition_sql = f"""
+        ALTER TABLE alert_history
+        PARTITION BY RANGE ({partition_expression}) (
+            PARTITION p{partition_name} VALUES LESS THAN ({partition_value})
+        );
```

Comment:
I would move it to a separate function, e.g. `create_partition_query`, to keep it more contained and concise

Reply:
@TomerShor moved partly to alert history crud. As SQL part should still be there as it's specific to initial migration..